LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/BasicsStation/Processors/CupsProtocolMessageProcessor.cs

```diff
@@ -0,0 +1,157 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoRaWan.NetworkServer.BasicsStation.Processors
+{
+    using System;
+    using System.Buffers;
+    using System.Buffers.Binary;
+    using System.IO;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using LoRaTools.CommonAPI;
+    using LoRaWan.NetworkServer.BasicsStation.JsonHandlers;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.Extensions.Logging;
+
+    internal class CupsProtocolMessageProcessor : ICupsProtocolMessageProcessor
+    {
+        private readonly IBasicsStationConfigurationService basicsStationConfigurationService;
+        private readonly LoRaDeviceAPIServiceBase deviceAPIServiceBase;
+        private readonly ILogger<CupsProtocolMessageProcessor> logger;
+
+        public CupsProtocolMessageProcessor(IBasicsStationConfigurationService basicsStationConfigurationService,
+                                            LoRaDeviceAPIServiceBase deviceAPIServiceBase,
+                                            ILogger<CupsProtocolMessageProcessor> logger)
+        {
+            this.basicsStationConfigurationService = basicsStationConfigurationService;
+            this.deviceAPIServiceBase = deviceAPIServiceBase;
+            this.logger = logger;
+        }
+
+        public async Task HandleUpdateInfoAsync(HttpContext httpContext, CancellationToken token)
+        {
+            using var reader = new StreamReader(httpContext.Request.Body);
+            var input = await reader.ReadToEndAsync();
```

Comment:
What's your suggestion here? Should we specify a maximum payload size?
The payload I am receiving from the BasicStation is now around 250bytes, but it's missing any "key" for Firmware Upgrade. Should we just limit it to 1K?

Reply:
I would define a reasonable upper limit - depending on the size of the "key" for the Firmware upgrade. It can even be 2k, but we should limit it and stop processing when we get over the limit.

Reply:
Could you please check changes in https://github.com/Azure/iotedge-lorawan-starterkit/pull/939/commits/43f2210150b205c5bfca161c52585373b476ca89 ? @p-schuler